seminar9/log.py

```diff
@@ -0,0 +1,7 @@
+from telegram import Update
+from datetime import datetime
+def log(update: Update, result):
+    dt = str(datetime.now())
+    output_string = ' '.join([dt, str(update.effective_user), update.message.text, result, '\n'])
+    with open('log.txt', mode='a') as file:
+        file.write(output_string)
```

Comment:
У вас логи записываются в одну строчку. Нужно сделать так, чтобы писались в столбик.
```suggestion
        file.write(output_string + '\n')
```

Reply:
Символ перевода на новую строку цепляется к логируемым данным через join. Или лучше так не делать?

Reply:
> Символ перевода на новую строку цепляется к логируемым данным через join. Или лучше так не делать?

Я вас не понял

Reply:
У меня логирование данных сделано так:
output_string = ' '.join([dt, str(update.effective_user), update.message.text, result, '\n'])
То есть я добавляю символ '\n' в конец списка из логируемых данных а потом склеиваю эту строку. Потом пишу в файл. То есть лог в итоге пишется в столбик.

Reply:
> У меня логирование данных сделано так: output_string = ' '.join([dt, str(update.effective_user), update.message.text, result, '\n']) То есть я добавляю символ '\n' в конец списка из логируемых данных а потом склеиваю эту строку. Потом пишу в файл. То есть лог в итоге пишется в столбик.

Странно. У меня лог файла записывался в одну строку.